tests/operators/functionals/test_l1_norm.py

```diff
@@ -0,0 +1,341 @@
+    torch.testing.assert_close(l1_norm.forward(x)[0], forward_x, rtol=1e-3, atol=1e-3)
+    torch.testing.assert_close(l1_norm.prox(x, sigma=0.1)[0], prox, rtol=1e-3, atol=1e-3)
+    torch.testing.assert_close(l1_norm.prox_convex_conj(x, sigma=0.1)[0], prox_complex_conjugate, rtol=1e-3, atol=1e-3)
+
+
+@pytest.mark.parametrize(
+    (
+        'x',
+        'forward_x',
+        'prox',
+        'prox_complex_conjugate',
+    ),
+    [
+        (
+            torch.tensor([1.0, -1.0], dtype=torch.float32),
+            torch.tensor((1), dtype=torch.float32),
+            torch.tensor([0.9, -0.9], dtype=torch.complex64),
+            torch.tensor([0.95, -0.95], dtype=torch.complex64),
+        ),
+    ],
+)
+def test_l1_mixed_real_complex_x_real_target_complex(
+    x,
+    forward_x,
+    prox,
+    prox_complex_conjugate,
+):
+    """Test if L1 norm matches expected values."""
+    l1_norm = L1Norm(weight=1, target=torch.tensor([0.5 + 0j, -0.5 + 0j], dtype=torch.complex64), keepdim=False)
+    torch.testing.assert_close(l1_norm.forward(x)[0], forward_x, rtol=1e-3, atol=1e-3)
+    torch.testing.assert_close(l1_norm.prox(x, sigma=0.1)[0], prox, rtol=1e-3, atol=1e-3)
+    torch.testing.assert_close(l1_norm.prox_convex_conj(x, sigma=0.1)[0], prox_complex_conjugate, rtol=1e-3, atol=1e-3)
+
+
+@pytest.mark.parametrize(
+    (
+        'x',
+        'forward_x',
+        'prox',
+        'prox_complex_conjugate',
+    ),
+    [
+        (
+            torch.tensor([1.0 + 0j, -1.0 + 0j], dtype=torch.complex64),
+            torch.tensor((1), dtype=torch.float32),
+            torch.tensor([0.9, -0.9], dtype=torch.complex64),
+            torch.tensor([0.95, -0.95], dtype=torch.complex64),
+        ),
+    ],
+)
+def test_l1_mixed_real_complex_x_complex_target_real(
+    x,
+    forward_x,
+    prox,
+    prox_complex_conjugate,
+):
+    """Test if L1 norm matches expected values."""
+    l1_norm = L1Norm(weight=1, target=torch.tensor([0.5, -0.5], dtype=torch.float32), keepdim=False)
+    torch.testing.assert_close(l1_norm.forward(x)[0], forward_x, rtol=1e-3, atol=1e-3)
+    torch.testing.assert_close(l1_norm.prox(x, sigma=0.1)[0], prox, rtol=1e-3, atol=1e-3)
+    torch.testing.assert_close(l1_norm.prox_convex_conj(x, sigma=0.1)[0], prox_complex_conjugate, rtol=1e-3, atol=1e-3)
+
+
+@pytest.mark.parametrize(
+    (
+        'x',
+        'dim',
+        'keepdim',
+        'shape_forward_x',
+    ),
+    [
+        (
+            torch.rand(16, 8, 1, 160, 160),
+            (-2, -1),
+            True,
+            torch.Size([16, 8, 1, 1, 1]),
+        ),
+        (
+            torch.rand(16, 8, 1, 160, 160),
+            (-2, -1),
+            False,
+            torch.Size([16, 8, 1]),
+        ),
+        (
+            torch.rand(16, 8, 1, 160, 160),
+            None,
+            False,
+            torch.Size([]),
+        ),
+        (
+            torch.rand(16, 8, 1, 160, 160),
```

Comment:
dontuse torch rand, use the rng helper as done in the other tests.
this is important for reproducible tests.